gui/src/launcher.rs

```diff
@@ -42,71 +48,163 @@ impl Launcher {
+                                        .push(text("Install Liana on another network")),
+                                )
+                                .on_press(ViewMessage::StartInstall)
+                                .padding(10)
+                                .width(Length::Fill)
+                                .style(button::Style::TransparentBorder.into()),
+                            ),
+                    )
+                    .max_width(500)
+                    .align_items(Alignment::Center),
+            )
+            .width(Length::Fill)
+            .height(Length::Fill)
+            .center_x()
+            .center_y(),
         )
-        .width(Length::Fill)
-        .height(Length::Fill)
-        .center_x()
-        .center_y()
-        .into()
+        .map(Message::View)
     }
 }
 
-#[derive(Debug, Clone)]
+#[derive(Debug)]
 pub enum Message {
-    Install,
-    Run(Network),
+    View(ViewMessage),
+    Install(PathBuf),
+    Checked(Result<Network, String>),
+    Run(PathBuf, app::config::Config, liana::config::Config),
+}
+
+#[derive(Debug, Clone)]
+pub enum ViewMessage {
+    StartInstall,
+    Check(Network),
+}
+
+async fn check_network_datadir(mut path: PathBuf, network: Network) -> Result<Network, String> {
+    path.push(network.to_string());
+    path.push(app::config::DEFAULT_FILE_NAME);
+
+    let cfg = app::Config::from_file(&path).map_err(|_| {
+        format!(
+            "Failed to read GUI configuration file in the directory: {}",
+            path.to_string_lossy()
+        )
+    })?;
+
+    liana::config::Config::from_file(Some(cfg.daemon_config_path.clone())).map_err(|e| match e {
+        ConfigError::FileNotFound
+        | ConfigError::DatadirNotFound => {
+            format!(
+                "Failed to read daemon configuration file in the directory: {}",
+                cfg.daemon_config_path.to_string_lossy()
+            )
+        }
+        ConfigError::ReadingFile(e) => {
+            if e.starts_with("Parsing configuration file: Error parsing descriptor") {
+                "There is an issue with the configuration for this network. You most likely use a descriptor containing one or more public key(s) without origin. Liana v0.2 and later only support public keys with origins. Please migrate your funds using Liana v0.1.".to_string()
+            } else {
+                format!(
+                    "Failed to read daemon configuration file in the directory: {}",
+                    cfg.daemon_config_path.to_string_lossy()
+                )
+            }
+        }
+        ConfigError::UnexpectedDescriptor(_) => {
+            "There is an issue with the configuration for this network. You most likely use a descriptor containing one or more public key(s) without origin. Liana v0.2 and later only support public keys with origins. Please migrate your funds using Liana v0.1.".to_string()
+        }
```

Comment:
This should be removed from the daemon in the future. It is never returned at the moment. (Do not modify this code though)